Sources/MapboxNavigation/CarPlayManager.swift

```diff
@@ -505,6 +505,7 @@ extension CarPlayManager {
      */
     public func previewRoutes(between waypoints: [Waypoint], completionHandler: @escaping CompletionHandler) {
         let options = NavigationRouteOptions(waypoints: waypoints)
+        options.distanceMeasurementSystem = NavigationSettings.shared.distanceUnit.measurementSystem
```

Comment:
It’s actually backwards that MapboxNavigationService and NavigationViewController tell NavigationSettings what locale settings to use. NavigationSettings is a singleton intended to apply consistently across the whole application, but allowing individual instances of other classes to set this one global value risks race conditions.

The order here is more sensible. What if we go one step further and push this line into NavigationRouteOptions itself?

Reply:
Makes total sense!